routers/web/auth/oauth.go

```diff
@@ -847,6 +847,11 @@ func SignInOAuth(ctx *context.Context) {
 		return
 	}
 
+	redirectTo := ctx.FormString("redirect_to")
+	if len(redirectTo) > 0 {
+		middleware.SetRedirectToCookie(ctx.Resp, redirectTo)
```

Comment:
for anyone else reviewing this, the same logic is in the openid auth provider https://github.com/go-gitea/gitea/blob/9cc15d18dfe25f5e0a7569ffb6203e9a4dbb2404/routers/web/auth/openid.go#L46

Reply:
Is there a need for the `else` clause in the code you linked, here as well?